lib/misc/msgList.c: missing #ifdef

This macro checks for BSD style printf(), which is not present
when compiling for uClibc. The linked functions are unnecessary in
this case, and they break compilation.

Signed-off-by: Karoly Kasza <kaszak@gmail.com>

--- openvmtools-stable-9.10.0.orig/open-vm-tools/lib/misc/msgList.c	2015-06-17 10:01:00.000000000 +0200
+++ openvmtools-stable-9.10.0/open-vm-tools/lib/misc/msgList.c		2015-06-17 10:01:00.000000000 +0200
@@ -487,6 +487,7 @@
    return messages->id;
 }
 
+#ifdef HAS_BSD_PRINTF
 
 /*
  *----------------------------------------------------------------------
@@ -566,6 +567,7 @@
    }
 }
 
+#endif
 
 /*
  *----------------------------------------------------------------------
